modules/disk-uploader/pkg/image/image.go

```diff
@@ -16,7 +17,24 @@ import (
 	tar "kubevirt.io/containerdisks/pkg/build"
 )
 
-func Build(diskPath string) (v1.Image, error) {
+const (
+	defaultInstanceType = "INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE"
+	defaultPreference   = "INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE"
+)
+
+func DefaultConfig(labels map[string]string) v1.Config {
```

Comment:
I'd suggest to use regexes/replacements here. One to filter the labels we want, and another one to convert the label name to env variable.

See for matching regex:
https://github.com/kubevirt/containerized-data-importer/blob/1f34e0ea9d5d5baa00478c4c466f3c451e84c557/pkg/controller/common/util.go#L386

Create inverse of this:
https://github.com/kubevirt/containerized-data-importer/blob/1f34e0ea9d5d5baa00478c4c466f3c451e84c557/pkg/importer/util.go#L77

Reply:
I don't think it is more or less complicated than adding a `filterLabels` func.

Reply:
Is the https://github.com/kubevirt/containerized-data-importer/blob/1f34e0ea9d5d5baa00478c4c466f3c451e84c557/pkg/importer/util.go#L77 function used for all labels/annotations?

Reply:
`envToLabel` is used in CDI when importing from a registry source. It only converts labels.

Reply:
> I don't think it is more or less complicated than adding a filterLabels func.

The `filterLabels` function was removed. Thanks for sharing CDI functions, it's useful to get famliar with them in that scenario.

Can you please help me to understand what labels needed except the 2 mentioned here? I don't think we should set more labels than what we need in the image config below:

```
func DefaultConfig(labels map[string]string) v1.Config {
	var env []string

	if value, exists := labels[defaultInstanceTypeLabel]; exists {
		env = append(env, fmt.Sprintf("%s=%s", defaultInstanceTypeEnv, value))
	}
	if value, exists := labels[defaultPreferenceLabel]; exists {
		env = append(env, fmt.Sprintf("%s=%s", defaultPreferenceEnv, value))
	}
	return v1.Config{Env: env}
}
```

Reply:
As Lee mentioned, it should at least also look after the `-kind` variants of these labels.

Reply:
Yes I saw it. I've modified it in this [revision](https://github.com/kubevirt/kubevirt-tekton-tasks/compare/fdc80d265fe74f103a1ce3fe8e2442fc3366b58a..a3b9ff76d4161b586c65313e23d4743577f1829e). Can you please take a look?

Reply:
@0xFelix @lyarwood Are you okay with current implementation? Can we resolve this discussion?